apollo-router/src/configuration/expansion.rs

```diff
@@ -81,16 +97,27 @@ impl Expansion {
             .map(|mode| mode.trim().to_string())
             .collect::<Vec<String>>();
 
-        let dev_mode_defaults = if std::env::var(APOLLO_ROUTER_DEV_ENV).ok().as_deref()
-            == Some("true")
+        let dev_mode_defaults = if crate::executable::APOLLO_ROUTER_DEV_MODE.load(Ordering::Relaxed)
         {
             tracing::info!("Running with *development* mode settings which facilitate development experience (e.g., introspection enabled)");
             dev_mode_defaults()
         } else {
             Vec::new()
         };
 
-        Ok(Expansion::builder()
+        let builder = Expansion::builder();
+        #[cfg(test)]
+        let builder = builder.mocked_env_vars(mocked_env_vars);
+        let listen_override = Override::builder()
+            .config_path("supergraph.listen")
+            .value_type(ValueType::String);
+        let listen = *crate::executable::APOLLO_ROUTER_LISTEN_ADDRESS.lock();
+        let listen_override = if let Some(listen) = listen {
+            listen_override.value(listen.to_string()).build()
+        } else {
+            listen_override.build()
+        };
```

Comment:
Consider using a slightly simpler version:
```suggestion
        let mut listen_override = Override::builder()
            .config_path("supergraph.listen")
            .value_type(ValueType::String);
        if let Some(listen) = *crate::executable::APOLLO_ROUTER_LISTEN_ADDRESS.lock() {
            listen_override.value = listen_override.value(listen.to_string());
        }
        listen_override.build()
```

Reply:
The return type of `Override::builder()` is not `Override` but a struct generated by the [`buildstructor`](https://crates.io/crates/buildstructor) proc macro crate. The builder type does not have public fields:

```rust
error[E0615]: attempted to take value of method `value` on type `configuration::expansion::__override_new_builder::__OverrideBuilder<(configuration::expansion::__override_new_builder::__Set<std::string::String>, configuration::expansion::__override_new_builder::__Optional<std::option::Option<std::string::String>>, configuration::expansion::__override_new_builder::__Optional<std::option::Option<serde_json::Value>>, configuration::expansion::__override_new_builder::__Set<configuration::expansion::ValueType>)>`
   --> apollo-router/src/configuration/expansion.rs:124:29
    |
124 |             listen_override.value = listen_override.value(listen.to_string());
    |                             ^^^^^ method, not a field
    |
    = help: methods are immutable and cannot be assigned to